### ISSUE TITLE
using axarray.concat meetting problem with 'numpy.ndarray' has no attribute '_concat_same_type'

### ISSUE THREAD

**Issue by CapitalLiu** [pydata/xarray#9851]
### What happened?

Numpy 2.1.3 Xarray 24.3.0
`Traceback (most recent call last):
  File "/data/guyida/graphcast/Demo/Inference.py", line 282, in <module>
    input_batch = xarray.concat([example_batch,sample_batch_new], dim='time')
  File "/usr/local/lib/python3.10/dist-packages/xarray/core/concat.py", line 277, in concat
    return _dataset_concat(
  File "/usr/local/lib/python3.10/dist-packages/xarray/core/concat.py", line 669, in _dataset_concat
    combined_var = concat_vars(
  File "/usr/local/lib/python3.10/dist-packages/xarray/core/variable.py", line 3050, in concat
    return Variable.concat(variables, dim, positions, shortcut, combine_attrs)
  File "/usr/local/lib/python3.10/dist-packages/xarray/core/variable.py", line 1782, in concat
    data = duck_array_ops.concatenate(arrays, axis=axis)
  File "/usr/local/lib/python3.10/dist-packages/xarray/core/duck_array_ops.py", line 391, in concatenate
    return _concatenate(as_shared_dtype(arrays), axis=axis)
  File "/usr/local/lib/python3.10/dist-packages/xarray/core/extension_array.py", line 100, in __array_function__
    res = HANDLED_EXTENSION_ARRAY_FUNCTIONS[func](*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/xarray/core/extension_array.py", line 48, in __extension_duck_array__concatenate
    return type(arrays[0])._concat_same_type(arrays)  # type: ignore[attr-defined]
AttributeError: type object 'numpy.ndarray' has no attribute '_concat_same_type'`

### MVCE confirmation

- [ ] Minimal example — the example is as focused as reasonably possible to demonstrate the underlying issue in xarray.
- [ ] Complete example — the example is self-contained, including all data and the text of any traceback.
- [ ] Verifiable example — the example copy & pastes into an IPython prompt or [Binder notebook](https://mybinder.org/v2/gh/pydata/xarray/main?urlpath=lab/tree/doc/examples/blank_template.ipynb), returning the result.
- [ ] New issue — a search of GitHub Issues suggests this is not a duplicate.
- [ ] Recent environment — the issue occurs with the latest version of xarray and its dependencies.

### Environment

<details>
Name: xarray
Version: 2024.3.0
Summary: N-D labeled arrays and datasets in Python
Home-page: https://xarray.dev/
Author: 
Author-email: xarray Developers <xarray@googlegroups.com>
License: Apache-2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: numpy, packaging, pandas
Required-by: graphcast

numpy:2.1.3

</details>

**Comment by keewis** [comment]
`xarray<2024.07.0` does not support `numpy>=2`. Thus, the only thing we can tell you is to upgrade to a newer version of `xarray` (at least `2024.07.0`) or to downgrade to `numpy<2`.
